exercises/IaC/ansible/conditionals/.templates/README.md.erb

```diff
@@ -0,0 +1,363 @@
+
+```YAML
+<%=
+hash_to_list=<<YAML
+- name: Hash to list example
+  hosts: all
+  vars:
+    user_groups:
+      Admin:
+      - user1
+      - user2
+      Team:
+      - user3
+      - user4
+      - user5
+
+  tasks:
+  - name: 'create user'
+    debug:
+      msg: "Creating user: {{item['key']}} in group: {{item['value']}}"
+    loop: "{{ user_groups| dict2items  }}"
+YAML
+%>
+```
+
+In the case of the above playbook `dict2items` does the following:
+
+```YAML
+user_groups:
+  admin:
+  - user1
+  - user2
+  team:
+  - user3
+  - user4
+  - user5
+```
+becomes
+
+```YAML
+- key: admin
+  value: [ user1, user2 ]
+- key: team
+  value: [ user3, user4, user5]
+```
+write the playbook to `<%= hash_to_list_path = write_to_file('ansible/hash_to_list.yml', hash_to_list) %>` and Run it with: `<%= command("ansible-playbook #{hash_to_list_path}") %>`. This will output the following:
+```
+
+
+<%= last_command_output %>
+```
+
+As you can see, the `user_groups` variable has been successfully used within the loop.
+
+### Nested loops
+Putting one loop within another is also possible. In order to surround a group of tasks with a loop they must be imported using the `include_tasks` plugin. The tasks below doing the following:
+
+- print a message to say that the group defined in the `group_name` variable is being created
+- print loop through the list defined in the `users` variable and print a message for item in it.
+
+```YAML
+
+<%=
+inner_tasks =<<YAML
+- name: "create group"
+  debug:
+    msg: "crearting group: {{group_name}}"
+
+- name: "create users"
+  debug:
+    msg: "creating user: {{item}}"
+  loop: "{{users}}"
+YAML
+%>
+```
+
+The following playbook then includes the tasks above and puts a `loop` condition against them.
+
+```YAML
+
+<%=
+outer_playbook =<<YAML
+---
+- name: super loop
+  hosts: all
+  vars:
+    user_groups:
+      Admin:
+      - user1
+      - user2
+      Team:
+      - user3
+      - user4
+      - user5
+
+  tasks:
+  - include: create_users_and_groups.yml group_name="{{item['key']}}" users="{{item['value']}}"
+    loop: "{{ user_groups| dict2items }}"
+YAML
+%>
+```
+
+The playbook starts with the same `user_groups` hash as the previous example so uses the `dict2items` filter to convert it. Each time around the loop the group name (`item['key']`) and and list of users (`item['value']`) are assigned to the variable names that are defined in the included tasks.
+
+Write the tasks to `<%= write_to_file('ansible/create_users_and_groups.yml', inner_tasks) %>` and the playbook to `<%= nested_loops_path = write_to_file('ansible/nested_loops.yml', outer_playbook) %>. Now run the playbook with: `<%= command("ansible-playbook #{nested_loops_path}") %>`. This will show the following:
```

Comment:
This line is rendering incorrectly in my markdown preview - Missing back-tick before '. Now run the playbook'